cloudant.go

```diff
@@ -53,12 +53,12 @@ type Index struct {
 }
 
 // NewClient ...
-func NewClient(username string, password string) (*Client, error) {
-	auth := couchdb.BasicAuth(username, password)
+func NewClient(username string, apikey string, password string) (*Client, error) {
+	auth := couchdb.BasicAuth(apikey, password)
```

Comment:
@Zambiorix I think this will be a bit confusing. For example, if I don't have an API key, the straightforward way would be passing in an empty string for the `apikey` parameter. What about this:

1. If `username == ""` and `apikey != ""`, use `auth := couchdb.BasicAuth(apikey, password)`.
1. If `username != ""` and `apikey == ""`, use `auth := couchdb.BasicAuth(username, password)`.
1. If both `username` and `apikey` are present, `apikey` takes priority.